test/mockBackend/mockAxios.js

```diff
@@ -0,0 +1,160 @@
+import MockAdapter from 'axios-mock-adapter'
+import { pathToRegexp } from 'path-to-regexp'
+import 'blob-polyfill'
+
+import axios from '@/base/api/axios'
+import { camelizeKeys, underscorizeKeys } from '@/utils/utils'
+
+import { ctx } from './index'
+
+// holds the current context for the mock axios
+/**
+ * @type {MockAdapter}
+ */
+let mockAxios
+
+export function initializeMockAxios () {
+  if (mockAxios) throw new Error('already have a mock!')
+  mockAxios = new MockAdapter(axios, { onNoMatch: 'throwException' })
+}
+
+export function resetMockAxios () {
+  if (mockAxios) {
+    // resets the state
+    mockAxios.reset()
+    // restores axios back to how it was (maybe enough on it's own?)
+    mockAxios.restore()
+  }
+  mockAxios = null
+}
+
+export function get (path, handler, options = {}) {
+  on('get', path, handler, options)
+}
+
+export function post (path, handler, options = {}) {
+  on('post', path, handler, options)
+}
+
+export function patch (path, handler, options = {}) {
+  on('patch', path, handler, options)
+}
+
+function on (method, path, handler, options = {}) {
+  const {
+    requireAuth = true,
+  } = options
+  const mockAxios = ensureMockAxios()
+  function onRequest (path) {
+    switch (method) {
+      case 'get': return mockAxios.onGet(path)
+      case 'post': return mockAxios.onPost(path)
+      case 'patch': return mockAxios.onPatch(path)
+      default: throw new Error('have not implemented method: ' + method)
+    }
+  }
+  // If we have a path with params/:like/:this/, we  match/extract them
+  // They are available to the handler as pathParams
+  const matcher = path.includes(':') ? createPathMatcher(path) : null
+  onRequest(matcher || path).reply(async config => {
+    if (!ctx.authUser && requireAuth) return notAuthenticated()
+    let data = config.data
+    let pathParams
+    if (matcher) {
+      pathParams = matcher.getParams(config.url)
+    }
+    if (['post', 'patch', 'put'].includes(config.method)) {
+      if (config.data instanceof FormData) {
+        // This is the reverse of toFormData (but ignoring the images)
+        data = camelizeKeys(JSON.parse(await config.data.get('document').text()))
+      }
+      else if (
+        typeof config.data === 'string' &&
+          config.headers['Content-Type'] === 'application/json'
+      ) {
+        data = camelizeKeys(JSON.parse(config.data))
+      }
+    }
+    const handlerResponse = handler({ ...config, data, pathParams })
+    if (!Array.isArray(handlerResponse)) throw new Error('mock handler must return an array')
+    const [statusCode, body] = handlerResponse
+    if (typeof statusCode !== 'number') throw new Error('mock handler array must have numeric status code as first arg')
+    return [statusCode, body && underscorizeKeys(body)]
+  })
+}
+
+export function cursorPaginated (path, getEntries, options = {}, makeResults = null) {
```

Comment:
`makeResults` could go into the `options`

Reply:
since `options` is passed as-is into mockAxios `get`, it feels a bit off to put other stuff inside...

Reply:
the `get` is only our `get` function, both `get` and `cursorPaginated` are part of our "mock axios" "library", so I think having a common object for the object fits, it's _our_ options object..

the only other use of the options is `{ requireAuth: false }`, it doesn't get passed into the axios mock adapter lib.

Reply:
makes sense now! thanks for the explanation, I changed the code